### PR TITLE
feat: Buy Bag events

### PR DESCRIPTION
**Pull request by gbugyis**
Adds events for NFT Buy Bag (adding to bag, Pay, Bag Update (unavailable/price change), Bag signed, Success screen)

### CODE REVIEW

File: src/analytics/constants.ts [hunk 1]
```diff
@@ -16,6 +16,12 @@ export enum EventName {
   NAVBAR_SEARCH_SELECTED = 'Navbar Search Selected',
   NAVBAR_SEARCH_EXITED = 'Navbar Search Exited',
   NFT_ACTIVITY_SELECTED = 'NFT Activity Selected',
+  NFT_BUY_ADDED = 'NFT Buy Bag Added',
+  NFT_BUY_BAG_CHANGED = 'NFT Buy Bag Changed',
+  NFT_BUY_BAG_PAY = 'NFT Buy Bag Pay Clicked',
+  NFT_BUY_BAG_REFUND = 'NFT Buy Bag Refund',
```
Comment (vm) [inline src/analytics/constants.ts:22]: Refunded

File: src/analytics/constants.ts [hunk 2]
```diff
@@ -16,6 +16,12 @@ export enum EventName {
...
+  NFT_BUY_BAG_SIGNED = 'NFT Buy Bag Signed',
+  NFT_BUY_BAG_SUCCESS = 'NFT Buy Bag Success',
```
Comment (vm) [inline src/analytics/constants.ts:24]: Succeeded

File: src/nft/components/bag/BagContent.tsx [hunk 3]
```diff
@@ -54,25 +57,47 @@ export const BagContent = () => {
...
+              usdPrice={fetchedPriceData}
+              clearUnavailableAssets={() => setItemsInBag(availableItems)}
+              didOpenUnavailableAssets={didOpenUnavailableAssets}
+              setDidOpenUnavailableAssets={setDidOpenUnavailableAssets}
+              isMobile={isMobile}
+            />
+          </Trace>
+        )}
+        {priceChangedAssets.length > 0 && (
+          <Trace
```
Comment (vm) [inline src/nft/components/bag/BagContent.tsx:80]: would this be better to do with a useeffect?

File: src/nft/components/bag/BagFooter.tsx [hunk 4]
```diff
@@ -48,6 +50,7 @@ interface BagFooterProps {
   bagStatus: BagStatus
   fetchAssets: () => void
   assetsAreInReview: boolean
+  eventProperties: Record<string, unknown>
```
Comment (vm) [inline src/nft/components/bag/BagFooter.tsx:53]: i think we would ideally type this, but if we're not doing that elsewhere then it's okay

**Review by vm — APPROVED** [review]
I think if we are okay with using Trace for modals then what we have works. Rest are nits.

File: src/nft/components/bag/BagContent.tsx (hunk 3 above)
Comment (gbugyis) [reply src/nft/components/bag/BagContent.tsx]: yes, good call

**Pull request merged**